### PR TITLE
[TIMOB-10787] iOS: Expose modern UUID-replacements

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-10787

### CODE REVIEW

File: iphone/Classes/PlatformModule.m [hunk 1]
```diff
@@ -227,6 +231,18 @@ -(NSString*)macaddress
...
 
+- (NSString *)identifierForVendor
+{
+    return [[[UIDevice currentDevice] identifierForVendor] UUIDString];
+}
+
+#if defined(USE_TI_PLATFORMIDENTIFIERFORADVERTISING) || defined(USE_TI_PLATFORMGETIDENTIFIERFORADVERTISING)
+- (NSString *)identifierForAdvertising
+{
+    return [[[ASIdentifierManager sharedManager] advertisingIdentifier] UUIDString];
```
Comment (vijaysingh-axway) [inline iphone/Classes/PlatformModule.m]: Apple  [here](https://developer.apple.com/documentation/adsupport/asidentifiermanager) suggest, we should use this by checking isAdvertisingTrackingEnabled. We can expose "isAdvertisingTrackingEnabled" property as well . Ygor Lemos has also suggested the same in comment.

Comment (hansemannn) [inline iphone/Classes/PlatformModule.m]: Good idea, added it!

File: iphone/Classes/PlatformModule.m [hunk 2]
```diff
@@ -227,6 +231,23 @@ -(NSString*)macaddress
     return [TiUtils appIdentifier];
 }
 
+- (NSString *)identifierForVendor
+{
+    return [[[UIDevice currentDevice] identifierForVendor] UUIDString];
+}
+
+#if defined(USE_TI_PLATFORMIDENTIFIERFORADVERTISING) || defined(USE_TI_PLATFORMGETIDENTIFIERFORADVERTISING)
+- (NSNumber *)isAdvertisingTrackingEnabled
```
Comment (vijaysingh-axway) [inline iphone/Classes/PlatformModule.m]: Make "isAdvertisingTrackingEnabled" property in .h file similar to 
+@property(readonly,nonatomic) NSString *identifierForAdvertising;

**Comment by vijaysingh-axway** [comment]
@hansemannn Can you resolve conflicts please.

**Comment by hansemannn** [comment]
@vijaysingh-axway Restored.

**Review by vijaysingh-axway — APPROVED** [review]
CR and FT passed.

**Comment by ewieberappc** [comment]
FR Passed. IdentifierForVendor and IdentifierForAdvertising return the appropriate values. Tested on multiple sims/devices with different apps and with advertising tracking enabled and disabled.

**Pull request merged**